InfiniTAM/Engine/RosPoseSourceEngine.cpp

```diff
@@ -0,0 +1,191 @@
+            + tf_world_to_camera_transform_current_.getOrigin();
+
+    tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame =
+        tf_world_to_camera_transform_at_start_.getBasis()
+            * tf_infinitam_origin_to_camera_current_translation_in_world_frame;
+
+    // Camera transform wrt infinitam origin in RVIZ.
+    tf_infinitam_origin_to_camera_transform_relative_.setOrigin(
+        tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame);
+    tf_infinitam_origin_to_camera_transform_relative_.setBasis(
+        tf_infinitam_origin_to_camera_current_rotation);
+
+    tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame_rotated =
+        tf_infinitam_origin_to_camera_current_rotation.inverse()
+            * tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame;
+
+    infinitam_translation_vector_.x =
+        -tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame_rotated
+            .getX();
+    infinitam_translation_vector_.y =
+        -tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame_rotated
+            .getY();
+    infinitam_translation_vector_.z =
+        -tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame_rotated
+            .getZ();
+    infinitam_rotation_matrix_.m00 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            0).getX();
+    infinitam_rotation_matrix_.m10 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            0).getY();
+    infinitam_rotation_matrix_.m20 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            0).getZ();
+    infinitam_rotation_matrix_.m01 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            1).getX();
+    infinitam_rotation_matrix_.m11 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            1).getY();
+    infinitam_rotation_matrix_.m21 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            1).getZ();
+    infinitam_rotation_matrix_.m02 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            2).getX();
+    infinitam_rotation_matrix_.m12 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            2).getY();
+    infinitam_rotation_matrix_.m22 =
+        tf_infinitam_origin_to_camera_transform_relative_.getBasis().getColumn(
+            2).getZ();
+
+    ROS_DEBUG_STREAM("\n transformation infinitam origin to camera pose: \n"
+              << infinitam_translation_vector_ << "\n");
+    ROS_DEBUG_STREAM("\n transformation infinitam origin to camera pose: \n"
+        << infinitam_rotation_matrix_ << "\n");
+
+    // if one wants to visualize the transformations.
+    if (broadcast_transformations) {
+      // origin of the map for infinitam.
```

Comment:
Comments should always be complete sentences and start with a capital letter and end with a period (and contain a verb as well). Use them wisely ;)